spec.html

```diff
@@ -35961,15 +36216,33 @@ <h1>
         </emu-alg>
         <emu-grammar>Atom :: `.`</emu-grammar>
         <emu-alg>
-          1. Let _A_ be the CharSet of all characters.
+          1. Let _A_ be AllCharacters(_rer_).
           1. If _rer_.[[DotAll]] is not *true*, then
             1. Remove from _A_ all characters corresponding to a code point on the right-hand side of the |LineTerminator| production.
           1. Return CharacterSetMatcher(_rer_, _A_, *false*, _direction_).
         </emu-alg>
         <emu-grammar>Atom :: CharacterClass</emu-grammar>
         <emu-alg>
           1. Let _cc_ be CompileCharacterClass of |CharacterClass| with argument _rer_.
-          1. Return CharacterSetMatcher(_rer_, _cc_.[[CharSet]], _cc_.[[Invert]], _direction_).
+          1. Let _cs_ be _cc_.[[CharSet]].
+          1. If _rer_.[[UnicodeSets]] is *false*, or if every element of _cs_ consists of a single character, return CharacterSetMatcher(_rer_, _cs_, _cc_.[[Invert]], _direction_).
+          1. Assert: _cc_.[[Invert]] is *false*.
+          1. Let _lm_ be an empty List of Matchers.
+          1. For each string _s_ in _cs_ whose length is greater than 1, iterating in descending order of string length, do
+            1. Let _cs2_ be a one-element CharSet containing the last code point of _s_.
+            1. Let _m2_ be the result of CharacterSetMatcher(_rer_, _cs2_, *false*, _direction_).
```

Comment:
Nit:

```suggestion
            1. Let _m2_ be CharacterSetMatcher(_rer_, _cs2_, *false*, _direction_).
```
Ditto everywhere below.

Reply:
Done in https://github.com/mathiasbynens/ecma262/pull/27